sys/vm_map.c

```diff
@@ -245,9 +241,8 @@ vm_map_t *vm_map_clone(vm_map_t *map) {
 
   rw_scoped_enter(&map->rwlock, RW_READER);
 
-  /* Temporarily switch to the new map, so that we may write contents. Note that
-     it's okay if we get preempted - the working vm map will be restored on
-     context switch. */
+  /* Temporarily switch to the new map, so that we may write contents. */
+  critical_enter();
```

Comment:
Ok. So the minimum assumption we must take is that we enter `vm_map_clone` with:
* user-space thread (i.e. with`p_uspace` in place),
* the only thread in a process – thus we know that no other thread (within the same process) will be started in the meantime.

If that is correct, please drop critical section and enforce above assumptions with `assert`.